kong/config_service.go

```diff
@@ -0,0 +1,49 @@
+package kong
+
+import (
+	"context"
+	"fmt"
+	"io"
+)
+
+// AbstractConfigService handles Config in Kong.
+type AbstractConfigService interface {
+	ReloadDeclarativeConfig(ctx context.Context, config any, checkHash int) error
```

Comment:
I'd suggest making `checkHas` a boolean. Integer allows a user to pass an arbitrary number while it doesn't really make sense in such a high-level API I think - it can be hidden as a detail of the HTTP API call.